newrelic/resource_newrelic_workload.go

```diff
@@ -0,0 +1,327 @@
+							Computed: true,
+						},
+					},
+				},
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The moment when the object was created, represented in milliseconds since the Unix epoch.",
+			},
+			"created_by": userReference,
+			"entity": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Description: "A list of entities manually assigned to this workload.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"guid": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+							Description: "The unique entity identifier in New Relic.",
+						},
+					},
+				},
+			},
+			"entity_search_query": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "A list of search queries that define a dynamic workload.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name of the query.",
+						},
+						"query": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The query.",
+						},
+					},
+				},
+			},
+			"guid": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique entity identifier of the workload in New Relic.",
+			},
+			"workload_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The unique entity identifier of the workload.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The workload's name.",
+			},
+			"permalink": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL of the Workload.",
+			},
+			"scope_accounts": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Description: "Accounts that will be used to get entities from.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"account_ids": {
+							Type:        schema.TypeSet,
+							Elem:        &schema.Schema{Type: schema.TypeInt},
+							Optional:    true,
+							MinItems:    1,
+							Description: "A list of accounts that will be used to get entities from.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceNewRelicWorkloadCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+	workload := expandWorkload(d)
+	accountID := d.Get("account.0.id").(int)
+
+	log.Printf("[INFO] Creating New Relic One workload %s", workload.Name)
+
+	createInput := workloads.CreateInput{
+		Name: workload.Name,
+	}
+
+	if len(workload.Entities) > 0 {
+		var entityGUIDs []string
+		for _, e := range workload.Entities {
+			entityGUIDs = append(entityGUIDs, *e.GUID)
+		}
+
+		createInput.EntityGUIDs = entityGUIDs
```

Comment:
It feels like we're doubling up the work here by first reading it into the workload struct and then copying those values into the createInput struct. Do you think it'd be better to just read the config straight into the inputStruct rather than going through another struct?

Reply:
👍